### PR TITLE
Support multiple domain names per tenant

### PR DESCRIPTION
**Pull request by orangewolf**
This PR allows a given tenant to be referred to by multiple domain names. For example samvera-repo.hykuup.com can be the same tenant as repo.samvera.org. This PR does not currently change the UI for adding or removing domain names. That will come in a later PR. It requires a migration but does not require a manual intervention on a tenant basis. . This work is also a precursor for a `is_public` feature that will allow tenants to be easily hidden behind HTTP auth for demos and pre-live accounts.

@samvera/hyku-code-reviewers

### CODE REVIEW

**Comment by orangewolf** [comment]
this is only failing do to a flaky feature spec - not due to any changed code here. it passes locally.

File: app/middleware/account_elevator.rb [hunk 1]
```diff
@@ -11,7 +11,8 @@ def parse_tenant_name(request)
   end
 
   def self.switch!(cname)
-    account = Account.find_by(cname: Account.canonical_cname(cname))
+    account = Account.joins(:domain_names).find_by(domain_names: { is_active: true,
```
Comment (smcfarlane) [inline app/middleware/account_elevator.rb:14]: you can use the `from_cname` class method here

**Review by smcfarlane — COMMENTED** [review]
Just commenting as I don't have a lot of context on Hyku but it looks good to me

File: app/middleware/account_elevator.rb (hunk 1 above)
Comment (orangewolf) [reply app/middleware/account_elevator.rb]: @smcfarlane good point - updated

**Review by bkiahstroud — APPROVED** [review]

**Pull request merged**